src/components/ha-wa-dialog.ts

```diff
@@ -0,0 +1,360 @@
+ * @see src/dialogs/more-info/ha-more-info-dialog.ts#L366
+ *
+ * @slot heading - Replace the entire header area.
+ * @slot navigationIcon - Leading header action (e.g. close/back button).
+ * @slot title - Header title. Click can toggle width if `width-on-title-click` is not "none".
+ * @slot subtitle - Header subtitle, shown under the title.
+ * @slot actionItems - Trailing header actions (e.g. buttons, menus).
+ * @slot - Dialog content body.
+ * @slot footer - Dialog footer content; typically action buttons.
+ *
+ * @csspart dialog - The dialog surface.
+ * @csspart header - The header container.
+ * @csspart body - The scrollable body container.
+ * @csspart footer - The footer container.
+ *
+ * @cssprop --dialog-content-padding - Padding for the dialog content sections. Defaults to 24px.
+ * @cssprop --ha-dialog-show-duration - Show animation duration. Defaults to 200ms.
+ * @cssprop --ha-dialog-hide-duration - Hide animation duration. Defaults to 200ms.
+ * @cssprop --ha-dialog-surface-background - Dialog background color. Defaults to surface.
+ * @cssprop --ha-dialog-border-radius - Border radius of the dialog surface. Defaults to 24px.
+ * @cssprop --dialog-z-index - Z-index for the dialog. Defaults to 8.
+ * @cssprop --dialog-surface-position - CSS position of the dialog surface. Defaults to relative.
+ * @cssprop --dialog-surface-margin-top - Top margin for the dialog surface. Defaults to auto.
+ * @cssprop --ha-dialog-expand-duration - Duration for width transitions when changing width. Defaults to 200ms.
+ *
+ * @attr {boolean} open - Controls the dialog open state.
+ * @attr {("small"|"medium"|"large"|"full")} width - Preferred dialog width preset. Defaults to "medium".
+ * @attr {("none"|"small"|"medium"|"large"|"full")} width-on-title-click - Target width when clicking the title. "none" disables.
+ * @attr {boolean} scrim-dismissable - Allows closing the dialog by clicking the scrim/overlay. Defaults to true.
+ * @attr {string} header-title - Header title text when no custom title slot is provided.
+ * @attr {string} header-subtitle - Header subtitle text when no custom subtitle slot is provided.
+ * @attr {boolean} flexcontent - Makes the dialog body a flex container for flexible layouts.
+ *
+ * @event opened - Fired when the dialog is shown.
+ * @event closed - Fired after the dialog is hidden.
+ */
+@customElement("ha-wa-dialog")
+export class HaWaDialog extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean, reflect: true })
+  public open = false;
+
+  @property({ type: String, reflect: true, attribute: "width" })
+  public width: DialogWidth = "medium";
+
+  @property({
+    type: String,
+    reflect: true,
+    attribute: "width-on-title-click",
+  })
+  public widthOnTitleClick: DialogWidthOnTitleClick = "none";
+
+  @property({ type: Boolean, reflect: true, attribute: "scrim-dismissable" })
+  public scrimDismissable = true;
```

Comment:
No not like this 🙂  boolean props should be always default false.

I would rename it. maybe `preventScrimClose`

Or we can introduce some kind of a `closeMode`: `fast` | `strict` 🤷